staging/src/k8s.io/client-go/tools/cache/reflector_watchlist_test.go

```diff
@@ -0,0 +1,441 @@
+				{SendInitialEvents: pointer.Bool(true), AllowWatchBookmarks: true, ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan, TimeoutSeconds: pointer.Int64(1)},
+			},
+			expectedStoreContent: []v1.Pod{*makePod("p1", "1")},
+		},
+		{
+			name:                  "check if stopping a watcher before sync results in creating a new watch-list request",
+			stopAfterWatchEvents:  1,
+			closeAfterWatchEvents: 3,
+			returnedWatchEvents: []watch.Event{
+				{Type: watch.Added, Object: makePod("p1", "1")},
+				// second request
+				{Type: watch.Added, Object: makePod("p2", "2")},
+				{Type: watch.Bookmark, Object: &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						ResourceVersion: "2",
+						Annotations:     map[string]string{"k8s.io/initial-events-end": "true"},
+					},
+				}},
+			},
+			expectedWatchRequests: 2,
+			expectedRequestOptions: []metav1.ListOptions{
+				{SendInitialEvents: pointer.Bool(true), AllowWatchBookmarks: true, ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan, TimeoutSeconds: pointer.Int64(1)},
+				{SendInitialEvents: pointer.Bool(true), AllowWatchBookmarks: true, ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan, TimeoutSeconds: pointer.Int64(1)},
+			},
+			expectedStoreContent: []v1.Pod{*makePod("p2", "2")},
+		},
+		{
+			name:                  "stopping a watcher after synchronization results in creating a new watch request",
+			stopAfterWatchEvents:  4,
+			closeAfterWatchEvents: 5,
+			returnedWatchEvents: []watch.Event{
+				{Type: watch.Added, Object: makePod("p1", "1")},
+				{Type: watch.Added, Object: makePod("p2", "2")},
+				{Type: watch.Bookmark, Object: &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						ResourceVersion: "2",
+						Annotations:     map[string]string{"k8s.io/initial-events-end": "true"},
+					},
+				}},
+				{Type: watch.Added, Object: makePod("p3", "3")},
+				// second request
+				{Type: watch.Added, Object: makePod("p4", "4")},
+			},
+			expectedWatchRequests: 2,
+			expectedRequestOptions: []metav1.ListOptions{
+				{SendInitialEvents: pointer.Bool(true), AllowWatchBookmarks: true, ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan, TimeoutSeconds: pointer.Int64(1)},
+				{AllowWatchBookmarks: true, ResourceVersion: "3", TimeoutSeconds: pointer.Int64(1)},
+			},
+			expectedStoreContent: []v1.Pod{*makePod("p1", "1"), *makePod("p2", "2"), *makePod("p3", "3"), *makePod("p4", "4")},
+		},
+		{
+			name: "expiring an established watcher results in returning an error from the reflector",
+			watchServerErrorPredicate: func() func(options metav1.ListOptions) error {
+				counter := 0
+				return func(options metav1.ListOptions) error {
+					counter++
+					if counter == 2 {
+						return apierrors.NewResourceExpired("rv already expired")
+					}
+					return nil
+				}
+			}(),
+			stopAfterWatchEvents:  3,
+			closeAfterWatchEvents: 5,
```

Comment:
I don't understand it - there aren't 5 events here...

Reply:
I must have copy&past it from the previous test cases. Will remove.
It didn't affect the test - was unused.